pkg/gameserversets/metrics.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gameserversets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	listerv1 "agones.dev/agones/pkg/client/listers/agones/v1"
+	mt "agones.dev/agones/pkg/metrics"
+	"agones.dev/agones/pkg/util/runtime"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	logger = runtime.NewLoggerWithSource("metrics")
+
+	keyName      = mt.MustTagKey("name")
+	keyNamespace = mt.MustTagKey("namespace")
+	keyFleetName = mt.MustTagKey("fleet_name")
+	keyType      = mt.MustTagKey("type")
+
+	gameServerCreationDuration = stats.Float64("gameserver_creation/duration", "The duration of gameserver creation", "s")
+)
+
+func init() {
+
+	stateViews := []*view.View{
+		{
+			Name:        "game_server_creation_duration",
```

Comment:
Does this metrics has a corresponding description in the `metrics.md` file? I may miss it.

Reply:
It's not and it's actually strange because this is the only metric that has an underscore in between for gameserver. I will change it to be `agones_gameserver_creation` to follow the same convention as the others as well and will add it to the `metrics.md`